src/main/java/ch/naviqore/raptor/router/TripMask.java

```diff
@@ -0,0 +1,11 @@
+package ch.naviqore.raptor.router;
+
+/**
+ * Represents a trip mask for a given day for a route.
+ *
+ * @param tripMask the trip mask for the day for the route, where each index represents trip (sorted by departure times)
+ *                 and the boolean value at that index indicates if the trip is taking place on the given day.
+ */
+public record TripMask(boolean[] tripMask) {
```

Comment:
I would not rename it to ServiceDayMaskProvider because it's still masking trips, also if we add more query params to our masks the ServiceDayMaskProvider would be the wrong name.

Rest I agree, see: 5f93d47